classes/container_actors.lua

```diff
@@ -538,10 +538,10 @@ end
 			if (bitBand(actorFlags, OBJECT_TYPE_PLAYER) ~= 0) then
 				if (not Details.ignore_nicktag) then
 					local actorNameAmbiguated = Ambiguate(actorName, "none")
-					actorObject.displayName = checkValidNickname(Details:GetNickname(actorNameAmbiguated, false, true), actorName) --defaults to player name
-					if (Details.remove_realm_from_name) then
-						actorObject.displayName = actorObject.displayName:gsub(("%-.*"), "")
-					end
+                    local nickname = Details:GetNickname(actorNameAmbiguated, false, true)
+                    if nickname then
+					    actorObject.displayName = checkValidNickname(nickname, actorName) --defaults to player name
+                    end
```

Comment:
Removed the `remove_realm_from_name` check here. If the user doesn't have a nickname then the displayName will be nil and it will hit the next block that sets the displayName and removes realms if needed.

Nicknames can't have special characters in them anyways.